server.js

```diff
@@ -38,168 +43,178 @@ const tracks = {
+	const reg = api.newContract(RegistrarABI, registryAddress);
+	const githubHintAddress = await reg.instance.getAddress.call({}, [githubHint, 'A']);
+
+	console.log(`Registering on GithubHint: ${sha3}, ${url}`);
+	// Should be this...
+	// api.newContract(GitHubHintABI, g).instance.hintURL.postTransaction({from: account.address}, [`0x${sha3}`, url]).then(() => {
+	// ...but will have to be this for now...
+	const hash = await sendTransaction(GitHubHintABI, githubHintAddress, 'hintURL', [`0x${sha3}`, url]);
+	console.log(`Transaction sent with hash: ${hash}`);
+
+	const operationsAddress = reg.instance.getAddress.call({}, [operationsContract, 'A']);
+	console.log(`Registering platform binary: ${commit}, ${platform}, ${sha3}`);
+	// Should be this...
+	// return api.newContract(OperationsABI, o).instance.addChecksum.postTransaction({from: account.address}, [`0x000000000000000000000000${commit}`, platform, `0x${sha3}`]);
+	// ...but will have to be this for now...
+	const hash2 = await sendTransaction(OperationsABI, operationsAddress, 'addChecksum', [`0x000000000000000000000000${commit}`, platform, `0x${sha3}`]);
+	console.log(`Transaction sent with hash: ${hash2}`);
+}));
 
 const server = app.listen(httpPort, function () {
 	const host = server.address().address;
 	const port = server.address().port;
 	console.log('push-release service listening at http://%s:%s', host, port);
 });
 
-const RegistrarABI = require('./res/registrar.json');
-const GitHubHintABI = require('./res/githubhint.json');
-const OperationsABI = require('./res/operations.json');
+function handleAsync (asyncFn) {
+	return (req, res) => asyncFn(req, res)
+		.then(() => {
+			if (!res.headersSent) {
+				throw new Error('No response from handler');
+			}
+		})
+		.catch(err => {
+			console.error(err);
+			res.status(400).end(`Error while processing the request:\n${err.toString()}\n`);
+		});
+}
+
+function fetchFile (commit, path) {
+	return new Promise((resolve, reject) => {
+		request.get({
+			headers: {
+				'User-Agent': githubRepo
+			},
+			url: `https://raw.githubusercontent.com/${githubRepo}/${commit}${path}`
+		}, function (error, response, body) {
+			if (error) {
+				reject(error);
+			} else {
+				resolve(body);
+			}
+		});
+	});
+}
+
+async function getNetwork () {
+	const n = await api.parity.netChain();
+	const network = (n === 'homestead' || n === 'mainnet' || n === 'foundation' ? 'foundation' : n.indexOf('kovan.json') !== -1 ? 'kovan' : n);
+	console.log(`On network ${network}`);
+	return network;
+}
+
+function isGoodTag (tag) {
+	return tag === 'nightly' || /^v[0-9]+\.[0-9]+\.[0-9]+$/.test(tag);
+}
+
+function sendTransaction (abi, address, method, args) {
+	let o = api.newContract(abi, address);
+	let tx = {
+		from: account.address,
+		to: address,
+		gasPrice: account.gasPrice,
+		data: o.getCallData(o.instance[method], {}, args)
+	};
+	return account.password === null
+		? api.eth.sendTransaction(tx)
```

Comment:
How will the transaction be signed in this case?

Reply:
We assume the node is unlocked.

Reply:
Can we thus check that the returned value is a TX hash? If the node is unlocked, it would return a request ID, right? It's just this kind of things that can be painful to debug

Reply:
Both (`eth_sendTransaction` and `personal_signAndSendTransaction`) methods always return `hash`. If the account is not unlocked it will probably hang idefinitely waiting for manual confirmation in the UI (which is also fine I believe).

Reply:
Ah right, I was thing of the `parity` one...